huff_utils/src/evm.rs

```diff
@@ -836,6 +836,146 @@ impl Opcode {
+            Opcode::Byte => (2, 1),
+            Opcode::Shl | Opcode::Shr | Opcode::Sar => (2, 1),
+            Opcode::Sha3 => (2, 1),
+            Opcode::Address => (0, 1),
+            Opcode::Balance => (1, 1),
+            Opcode::Origin => (0, 1),
+            Opcode::Caller => (0, 1),
+            Opcode::Callvalue => (0, 1),
+            Opcode::Calldataload => (1, 1),
+            Opcode::Calldatasize => (0, 1),
+            Opcode::Calldatacopy => (3, 0),
+            Opcode::Codesize => (0, 1),
+            Opcode::Codecopy => (3, 0),
+            Opcode::Gasprice => (0, 1),
+            Opcode::Extcodesize => (1, 1),
+            Opcode::Extcodecopy => (4, 0),
+            Opcode::Returndatasize => (0, 1),
+            Opcode::Returndatacopy => (3, 0),
+            Opcode::Extcodehash => (1, 1),
+            Opcode::Blockhash => (1, 1),
+            Opcode::Coinbase => (0, 1),
+            Opcode::Timestamp => (0, 1),
+            Opcode::Number => (0, 1),
+            Opcode::Difficulty => (0, 1),
+            Opcode::Prevrandao => (0, 1),
+            Opcode::Gaslimit => (0, 1),
+            Opcode::Chainid => (0, 1),
+            Opcode::Selfbalance => (0, 1),
+            Opcode::Basefee => (0, 1),
+            Opcode::Pop => (1, 0),
+            Opcode::Mload => (1, 1),
+            Opcode::Mstore | Opcode::Mstore8 => (2, 0),
+            Opcode::Sload => (1, 1),
+            Opcode::Sstore => (2, 0),
+            Opcode::Jump => (1, 0),
+            Opcode::Jumpi => (2, 0),
+            Opcode::Pc => (0, 1),
+            Opcode::Msize => (0, 1),
+            Opcode::Gas => (0, 1),
+            Opcode::Jumpdest => (0, 0),
+            Opcode::Push0 => (0, 1),
+            Opcode::Push1 => (0, 1),
+            Opcode::Push2 => (0, 1),
+            Opcode::Push3 => (0, 1),
+            Opcode::Push4 => (0, 1),
+            Opcode::Push5 => (0, 1),
+            Opcode::Push6 => (0, 1),
+            Opcode::Push7 => (0, 1),
+            Opcode::Push8 => (0, 1),
+            Opcode::Push9 => (0, 1),
+            Opcode::Push10 => (0, 1),
+            Opcode::Push11 => (0, 1),
+            Opcode::Push12 => (0, 1),
+            Opcode::Push13 => (0, 1),
+            Opcode::Push14 => (0, 1),
+            Opcode::Push15 => (0, 1),
+            Opcode::Push16 => (0, 1),
+            Opcode::Push17 => (0, 1),
+            Opcode::Push18 => (0, 1),
+            Opcode::Push19 => (0, 1),
+            Opcode::Push20 => (0, 1),
+            Opcode::Push21 => (0, 1),
+            Opcode::Push22 => (0, 1),
+            Opcode::Push23 => (0, 1),
+            Opcode::Push24 => (0, 1),
+            Opcode::Push25 => (0, 1),
+            Opcode::Push26 => (0, 1),
+            Opcode::Push27 => (0, 1),
+            Opcode::Push28 => (0, 1),
+            Opcode::Push29 => (0, 1),
+            Opcode::Push30 => (0, 1),
+            Opcode::Push31 => (0, 1),
+            Opcode::Push32 => (0, 1),
+            Opcode::Dup1 => (1, 2),
```

Comment:
these dup and swap look wrong

Reply:
agree, followed blindlly evm.codes :/

Reply:
force-pushed the fix

Reply:
@devtooligan was testing it against elohim contract and indeed we need those numbers
if inside macro I would first do swap1 then, I need to be sure that stack is having at least 2 elements